docs/usage.md

```diff
@@ -31,59 +41,133 @@ This part of the documentation will explain how to consume the connect api.
-      "read": true
-    }
-  },
-  "sessionToken": "r:xxxxxx"
+  --header 'x-parse-session-token: '$SESSION_TOKEN
+```
+
+See [Get object](#get-object) for a list of all methods and example response.
+
+### <a name="oauth-authentication">OAuth Authentication</a>
+
+In order to send items (i.e. create or update objects in classes), you need to authenticate each of your users with the Connect OAuth Flow, so you can send items "in their name". The Connect OAuth flow follow standard OAuth best practice, and works like a "connect with Facebook" or "login with Github" flow you probably already encountered somewhere.
+
+#### <a name="oauth-authentication-implementation">OAuth Authentication Implementation</a>
+
+- First, create or login to a Connect account, and then from the dashboard create an Application
+- Once your application is created, you'll obtain two keys:
+  - a `publicKey`, which is used in your app to initiate the OAuth flow
+  - a `secretKey`, which _should be kept secret_ and used only from your server (not the clientside app) at the end of the OAuth flow
+- Still in the dashboard, update your application to enter one or multiple `redirectUris` (separated by a comma). Declaring your `redirectUris` let Connect knows where we can redirect the user after they grant access to your app. It could be something like `https://yourwebsite.com/connectOAuthRedirect` or for mobile apps `deeplinkscheme://connectOAuthRedirect`
+- In your app, add a button somewhere (for example in your user's preference), that say "Link with Connect", and on click open a browser at the url `https://connect-project.io/authorize?client_id=[publicKey]&redirect_uri=[redirectUri]` (don't forget to replace the `publicKey` with yours, and the `redirectUri` with one from the list you declared in the dashboard)
+- After opening the previous url in a browser, the user is guided by Connect to login and grant access to your app
+- If the user grants you access, then we will redirect him to `[redirectUri]?authorization_code=[authorization_code]`
+- Your app should get the `authorization_code`, send it to your own server, which will perform the secure call to get the access token:
+
+```bash
+curl --location --request POST 'https://connect-project.io/oauth/token' \
+--header 'Content-Type: application/x-www-form-urlencoded' \
+--data-urlencode 'client_id='$publicKey \
+--data-urlencode 'client_secret='$secretKey \
+--data-urlencode 'grant_type=authorization_code' \
+--data-urlencode 'code='$authorization_code \
+--data-urlencode 'redirect_uri'=$redirectUri
+
+Response:
+{
+    "access_token": "XXXXXXXXXXXXXXXXXXXX",
+    "token_type": "Bearer",
+    "expires_in": 2591999,
+    "refresh_token": "YYYYYYYYYYYYYYYYYY",
+    "scope": []
 }
 ```
 
-At this point you can set the session token:
+To test it, there is a simple endpoint that get you the `userId` associated to this token:
 
 ```bash
-SESSION_TOKEN=r:xxxxxx
+GET localhost:1337/oauth/user "Authorization: Bearer $accessToken"
```

Comment:
done

Reply:
still not working due to slight problem on the `API_URL` variable. It was useful that it included `parse` when it could also be `parse-sandbox`, but now that it is always `parse` it should point to https://connect-project.io so it can also be used to link the oauth api. the variable could be renamed `CONNECT_URL` to reflect this change.

Reply:
fixed

Reply:
nice, thanks!